test/gtest/gtest_common.hpp

```diff
@@ -0,0 +1,164 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include <algorithm>
+#include <gtest/gtest.h>
+#include <iostream>
+#include <iterator>
+#include <miopen/env.hpp>
+#include <tuple>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "../driver.hpp"
+
+void default_check(const std::string& err) { std::cout << err; }
+
+void tuning_check(const std::string& err)
+{
+    // TEST_TUNING - the test should fail if output contains "Error" or "failed".
+    EXPECT_FALSE(err.find("Error") != std::string::npos || err.find("failed") != std::string::npos);
+    default_check(err);
+}
+
+enum class Gpu : int
+{
+    Default = 0,
```

Comment:
@CAHEK7 Rename to `none` to allow things like `disabled<Gpu::none>` which looks cleaner than `disabled<Gpu::Default>` (what is the default GPU?)